docs/root/configuration/observability/access_log/usage.rst

```diff
@@ -197,6 +197,16 @@ The following command operators are supported:
   In typed JSON logs, PROTOCOL will render the string ``"-"`` if the protocol is not
   available (e.g. in TCP logs).
 
+%STREAM_STATE%
```

Comment:
Sorry I missed this on the first pass but I think this is somewhat redundant with other existing access log fields, and stream info fields and probably something we don't want to represent as a single state int.

it's unclear what the status should be for TCP if the upstream connection isn't established.  meanwhile we should be able to infer that there's an upstream connection via upstream_host and upstream_cluster info. Is the goal here just to communicate if the request is complete?  If so I wonder if we can do something with the DURATION field to indicate that the last byte out is inconclusive.

Reply:
@alyssawilk the reason this change was introduced in first place, is overcoming the change on DURATION in this PR: https://github.com/envoyproxy/envoy/pull/26203.
Since it is now returning a value at any access log (also for periodic access logs), it couldn't be used as an indicator whether the stream has ended yet, or not.
Yes, the goal was communicating whether the request was complete or not, i.e distinguishing between end of stream access log and periodic access logs..

Reply:
I tend to agree that the implicit "started -> inprogress -> ended" state machine doesn't quite match the actual behavior of streams in Envoy. The reality, as @alyssawilk mentioned, is somewhat complicated and I'm apprehensive of presenting a picture which diverges from reality. 

I quite like the idea of using RESPONSE_FLAGS as an analog for this, and would be supportive of that approach. If we basically want the old-school DURATION behavior, I suppose we could just introduce a new token with those old semantics, too.

Reply:
@RyanTheOptimist I don't think going back to old DURATION behavior is beneficial. I actually make use of %STREAM_STATE%. Now that we have 3 possible occasions of access logs (start of connection/stream, periodic and the good old end of stream). The idea was that start of stream access log would return 'Started' for %STREAM_STATE%, periodic access logs would return 'InProgress', and end of stream will return 'Ended'. I completely understand your concerns about the current weird implementation using some kind of state machine. If there's a smarted way to achieve the same, I'm up for changing this, but for me at least, it's very important to keep these semantics for %STREAM_STATE%. Thanks to you both for looking into this, if we can agree on a better way to implement this, I'll do my best to fix this before upcoming release.

Reply:
I could see an arugment for an enum for access log type, which clearly indicated which form of access logging this was.  I don't think it should be STREAM_STATE in that case, and I don't think it would be a property of the stream info or the stream, but indicate the cause of the log getting kicked off.

Reply:
I tend to agree. Can we keep the values for %ACCESS_LOG_TYPE% to be Started,InProgress,Ended?

Reply:
I don't think so.  It's really not extensible - if we want a log before upstream connection establishment, you'd need something before "started" and I know @paul-r-gall is planning on adding a downstream log on response headers sent, which would mean there would be the awkwardness of having 2 in progress options,  inProgress and a StillInProgressButResponseHeadersSent ?
I think they should be named after the log trigger, so as we add new log triggers there's a clear naming pattern we can extend.

Reply:
@alyssawilk what names would you suggest?

Reply:
periodic SG. End is ambiguous (it could be end upstream, end downstream, end reading etc.)  as is start (start of downstream connection or start of upstream connection?)   did you want the upstream log when the upstream connection/stream is established, or when the upstream response occurs?  the issue indicates the former and your implementation indicates the latter and I'm not sure on naming until I know what the actual plan is.  If you actually wanted to do it on upstream connection/stream establishment maybe periodic, upstream_associated, downstream_termination?  @RyanTheOptimist is often better at naming and may have better ideas